### ISSUE TITLE
Is subsampling performed once for each iteration?

### ISSUE THREAD

**Issue by liujxing** [dmlc/xgboost#3242]
I am not sure from the documentation how subsampling is done here. Is subsampling done for each iteration, so the model has a different data for constructing each tree? Or is subsampling done only at the beginning of calling xgb.train, then the same data is used for constructing all trees?

**Comment by liujxing** [comment]
I just did an experiment of customizing a loss function which prints out the mean of label at each iteration, and the number is the same across iterations. So it seems xgboost uses the same data for each iteration.

**Comment by hcho3** [comment]
>Is subsampling done for each iteration, so the model has a different data for constructing each tree?

Yes. Subsampling is done for each iteration: https://github.com/dmlc/xgboost/blob/master/src/tree/updater_basemaker-inl.h#L145
